### DISCUSSION TITLE
Can harbor be extended to support Nuget, Python, and NPM packages?

### DISCUSSION THREAD

**Discussion by yemboado-ouoba** [goharbor/harbor#19381] [General]
I have a general question. Could (in theory) Harbor be extended to support NuGet, Python, NPM, Maven packages? For example, if someone were to fork the code, could he add functionality to support these packages? if yes, how difficult would it be?
Thanks,

**Comment by zyyw** [comment]
Hi @yemboado-ouoba, Harbor is designed as an artifact registry for OCI compatible artifacts. I am afraid NPM packages is not supported to be supported in Harbor, most likely Nuget and Python will not be supported either.

**Comment by Vad1mo** [comment]
in theory, yes, it could be done. It would involve numerous changes, thou.  Another more future-proof approach is to add OCI support to NuGet, Python, NPM and Maven. There are people working on that, search on OCI project on GitHub. eg. https://github.com/coreos/rpm-ostree/issues/4155, https://github.com/reproducible-containers/repro-get

**Comment by mkyc** [comment]
It was mentioned during recent KubeCon Europe. Support for maven would unblock migration from Nexus to Harbor.

**Comment by AmazingTurtle** [comment]
tbf it takes quite some steps to understand that harbor "only" supports containers and helm charts. repo about:

> An open source trusted cloud native registry project that stores, signs, and scans content.

Makes me think that harbor supports a wide variety of registries, such as npm or maven registries as well. Docs also unclear on that. Possibly this should be improved.
